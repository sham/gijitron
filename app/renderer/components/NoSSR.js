import React from 'react';
import PropTypes from 'prop-types';

export default class NoSSR extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      canRender: false
    };
  }
  
  componentDidMount() {
    this.setState({ canRender: true });
  }
  
  render() {
    const DefaultOnSSR = () => null;
    const { children, onSSR = <DefaultOnSSR /> } = this.props;
    const { canRender } = this.state;
  
    return canRender ? children : onSSR;
  }
}
NoSSR.propTypes = {
  children: PropTypes.element.isRequired,
  onSSR: PropTypes.element
};