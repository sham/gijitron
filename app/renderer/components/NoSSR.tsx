import React from 'react';

interface IProps {
  children: any;
  onSSR?: any;
}

interface IState {
  canRender: boolean;
}

export default class NoSSR extends React.Component<IProps, IState> {
  constructor(args) {
    super(args);
    this.state = {
      canRender: false
    };
  }

  public componentDidMount() {
    this.setState({ canRender: true });
  }

  public render() {
    const DefaultOnSSR = () => null;
    const { children, onSSR = <DefaultOnSSR /> } = this.props;
    const { canRender } = this.state;

    return canRender ? children : onSSR;
  }
}
