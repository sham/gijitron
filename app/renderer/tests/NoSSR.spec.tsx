import * as React from 'react'
import * as Renderer from 'react-test-renderer'

import NoSSR from '../components/NoSSR'

test('component NoSSR', () => {
  const component = Renderer.create(
    <NoSSR>
      <p>No SSR.</p>
    </NoSSR>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
