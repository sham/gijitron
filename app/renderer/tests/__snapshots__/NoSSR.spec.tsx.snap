// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component NoSSR 1`] = `
<p>
  No SSR.
</p>
`;
